### PR TITLE
Define props

### PR DESCRIPTION
**Pull request by rubyqorn**
Define undefined properties in PHPixie\Debug\Tracer\Trace\Element class

### CODE REVIEW

File: src/PHPixie/Debug.php [hunk 1]
```diff
@@ -4,8 +4,16 @@
...
 {
+    /**
+     * @var \PHPixie\Debug\Builder
+     */ 
     protected $builder;
     
+    /**
+     * Instance
+     * 
+     * @var object
```
Comment (dracony) [inline src/PHPixie/Debug.php:15]: This is wrong, as the instance type will be \PHPixie\Debug not just object

File: src/PHPixie/Debug.php [hunk 2]
```diff
@@ -4,8 +4,16 @@
 
 class Debug
 {
+    /**
+     * @var \PHPixie\Debug\Builder
+     */ 
     protected $builder;
     
+    /**
+     * Instance
```
Comment (dracony) [inline src/PHPixie/Debug.php:13]: I'm not sure this description which is basically the variable name adds any value

File: src/PHPixie/Debug.php [hunk 3]
```diff
@@ -14,21 +22,48 @@ public function __construct()
         static::$instance = $this;
     }
     
+    /**
+     * Instance of \PHPixie\Debug\Builder
```
Comment (dracony) [inline src/PHPixie/Debug.php:26]: Again, not sure if such descriptions are helpful, but won't comment on each.

File: src/PHPixie/Debug.php [hunk 4]
```diff
@@ -14,21 +22,48 @@ public function __construct()
...
+     * @param bool $exception 
+     * @param bool $error 
+     */ 
     public function registerHandlers($shutdownLog = false, $exception = true, $error = true)
     {
         $this->builder->handlers()->register($shutdownLog, $exception, $error);
     }
     
+    /**
+     * Displaying logged messages
```
Comment (dracony) [inline src/PHPixie/Debug.php:58]: A much better description would describe what the parameters do.

File: src/PHPixie/Debug.php [hunk 5]
```diff
@@ -39,6 +74,17 @@ public function dumpLog($withTitle = true, $withTraceArguments = true, $shortVal
         return $log;
     }
     
+    /**
+     * Displaying code where this method was
```
Comment (dracony) [inline src/PHPixie/Debug.php:78]: This method does not actually display anything, it returns a formatted string representing an exception.

File: src/PHPixie/Debug.php [hunk 6]
```diff
@@ -49,26 +95,57 @@ public function exceptionMessage($exception, $backtraceLimit = null, $neighborin
         return $message;
     }
     
+    /**
+     * Get instance of Trace class
```
Comment (dracony) [inline src/PHPixie/Debug.php:99]: Without an explanation of what the Trace is for it's kind of meaningless.

**Review by dracony — CHANGES_REQUESTED** [review]
Thanks a lot for your help!

I really appreciate you adding the classes to methods and properties. I did not comment on each line and instead have groups of feedback:

* The descriptions of methods have grammatical errors and are often misleading. A method that returns a formatted string was described as displaying it.
* The descriptions are not helpful in describing methods, and some like "Constructor" are just taking space.
* Some of the method arguments are wrong. e.g. backtraceLimit in one is types as "null". That's not the actual parameter type, just the default value. Again, it's just confusing to the reader.

Don't just add documentation wholesale just for having it there, like "Instance" or "Constructor". If something is self explanatory it does not need a description. I don't also know if there is much value in adding it to protected methods which the users won't actually call. 

My general advice would be, find the places that you feel actually need documentation and add a thorough one there. I really appreciate adding the return values though, this seems really helpful for IDEs.

Thanks!

**Comment by rubyqorn** [comment]
Thanks for feedback)

**Pull request closed without merging**